presale/models/presale_order.py

```diff
@@ -34,6 +35,8 @@ def action_validate(self):
             presale_order.state = 'confirmed'
             # Sales order creation
             order_lines = []
+            if presale_order.order_line_ids == []:
+                raise UserError(_('An empty order cannot be confirmed'))
```

Comment:
if you want to add this `UserError` then, the line `presale_order.state = 'confirmed'` should goes after this check.